cmd/manager/main.go

```diff
@@ -210,6 +209,7 @@ func main() {
 
 	if err = (&controllers.ClusterExtensionReconciler{
 		Client:                cl,
+		Reader:                mgr.GetAPIReader(),
```

Comment:
Was this an intentional decision to use an API only reader to make sure an updated certificate is being used?

I'm just wondering if it's required, because we would be using operator-controller's client configured with manager, instead of using the client with SA. (Not a change for now, if just wondering if an API only reader is required specifically instead of a cache backed one)

Reply:
Yes; if we use the non-Reader interface, it requires adding Secrets to the cache, which we really don't want to do. This was suggested by @joelanford